app/models/concerns/instance/course_components_concern.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+module Instance::CourseComponentsConcern
+  extend ActiveSupport::Concern
+  include CourseComponentQueryConcern
+
+  module ClassMethods
+    def available_components
+      @available_components ||= Course::ControllerComponentHost.components
+    end
+
+    def disableable_components
+      @disableable_components ||= available_components.select(&:can_be_disabled?)
+    end
+  end
+
+  delegate :available_components, to: :class
```

Comment:
Can you verify if there is a need for the `delegate` statement? From what I remember placing the class methods to be extended in a module `ClassMethods` within a concern does that automatically for you already.

Reply:
I defined the method at the class level because the method should give the same result for all instances (for the foreseeable future), i.e. we can call

```ruby
Instance.available_components
```

The delegation allows `available_components` to be called at the instance level:
```ruby
instance.available_components
```

If you value consistency/parity with `course/course_components_concern.rb` more, I can remove the class methods and just let them be instance methods.